src/qibojit/custom_operators/backends.py

```diff
@@ -192,28 +191,34 @@ def __init__(self):
                 ]
             }
 
-        # load gate kernels
+        # load core kernels
         kernels = []
         for kernel in self.KERNELS:
-            kernels.append(f"{kernel}_kernel{self.kernel_double_suffix}")
-            kernels.append(f"{kernel}_kernel{self.kernel_float_suffix}")
-            kernels.append(f"multicontrol_{kernel}_kernel{self.kernel_double_suffix}")
-            kernels.append(f"multicontrol_{kernel}_kernel{self.kernel_float_suffix}")
-        for ntargets in range(3, self.MAX_NUM_TARGETS+1):
-            kernels.append(f"apply_multi_qubit_gate_kernel{self.kernel_double_suffix[0:-2]}, {2**ntargets}>")
-            kernels.append(f"apply_multi_qubit_gate_kernel{self.kernel_float_suffix[0:-2]}, {2**ntargets}>")
-        kernels.append(f"collapse_state_kernel{self.kernel_double_suffix}")
-        kernels.append(f"collapse_state_kernel{self.kernel_float_suffix}")
-        kernels.append(f"initial_state_kernel{self.kernel_double_suffix}")
-        kernels.append(f"initial_state_kernel{self.kernel_float_suffix}")
+            kernels.append(f"{kernel}_kernel")
+            kernels.append(f"multicontrol_{kernel}_kernel")
+        kernels.append("collapse_state_kernel")
+        kernels.append("initial_state_kernel")
```

Comment:
Here we could avoid this loop (creating the `kernels` list) by loading the kernels and populating the `self.gates` dictionary directly. For example:
```Python
        # load core kernels
        self.gates = {}
        from qibojit.custom_operators import raw_kernels
        def kernel_loader(name, ktype):
            code = getattr(raw_kernels, name)
            code = code.replace("T", f"thrust::complex<{ktype}>")
            gate = cp.RawKernel(code, name, ("--std=c++11",))
            self.gates[f"{name}_{ktype}"] = gate

        for ktype in ("float", "double"):
            for name in self.KERNELS:
                kernel_loader(f"{name}_kernel", ktype)
                kernel_loader(f"multicontrol_{name}_kernel", ktype)
            kernel_loader("collapse_state_kernel", ktype)
            kernel_loader("initial_state_kernel", ktype)

        # load multiqubit kernels
        for ntargets in range(3, self.MAX_NUM_TARGETS + 1):
            for ktype in ("float", "double"):
                code = getattr(raw_kernels, "apply_multi_qubit_gate_kernel")
                code = code.replace("T", f"thrust::complex<{ktype}>")
                code = code.replace("nsubstates", str(2 ** ntargets))
                code = code.replace("MAX_BLOCK_SIZE", str(self.DEFAULT_BLOCK_SIZE))
                gate = cp.RawKernel(code, "apply_multi_qubit_gate_kernel", ("--std=c++11",))
                self.gates[f"apply_multi_qubit_gate_kernel_{ktype}_{ntargets}"] = gate
```
Feel free to keep whichever approach you prefer, given that none is significantly cleaner than the other.

Reply:
Approved, thanks.